### PR TITLE
Adopt go.work, rename modules to prep for monorepo

### PR DESCRIPTION
**Pull request by protolambda**
This PR:
- renames the modules to their final destination name in the monorepo
- splits the packages into modules: `op-node`, `op-proposer`, `op-batcher`, `op-e2e`
- moves `ops` to `ops-bedrock`, to prepare for monorepo
- Many Makefile and Dockerfile updates to make the above work
- updates CI (including getting coverage for all the separate modules)

### CODE REVIEW

**Comment by protolambda** [comment]
Stuck on a go issue:
```
../../../go/pkg/mod/github.com/btcsuite/btcd@v0.22.0-beta/chaincfg/genesis.go:10:2: ambiguous import: found package github.com/btcsuite/btcd/chaincfg/chainhash in multiple modules:
        github.com/btcsuite/btcd v0.22.0-beta (/home/proto/go/pkg/mod/github.com/btcsuite/btcd@v0.22.0-beta/chaincfg/chainhash)
        github.com/btcsuite/btcd/chaincfg/chainhash v1.0.0 (/home/proto/go/pkg/mod/github.com/btcsuite/btcd/chaincfg/chainhash@v1.0.0)
```

Which I think causes a build issue which the linting tooling can't handle (but go itself can somehow...):
```
golangci-lint run -E asciicheck,goimports,misspell --fix
WARN [runner] Can't run linter goanalysis_metalinter: buildir: failed to load package : could not load export data: no export data for "github.com/btcsuite/btcd/chaincfg/chainhash" 
ERRO Running error: 1 error occurred:
        * can't run linter goanalysis_metalinter: buildir: failed to load package : could not load export data: no export data for "github.com/btcsuite/btcd/chaincfg/chainhash"
```

File: Makefile [hunk 1]
```diff
@@ -18,9 +18,17 @@ submodules:
 	fi
 .PHONY: submodules
 
-opnode:
-	go build -o ./bin/op ./opnode/cmd
-.PHONY: opnode
+op-node:
+	cd ./op-opnode && make op-node
```
Comment (trianglesphere) [inline Makefile:22]: can use `make -C`

**Review by trianglesphere — COMMENTED** [review]
In general this looks good. The big thing I'd like is to set a cutoff date for which PRs go in first and which PRs should be built on top of this branch (to try to limit merge / rebase conflicts).

**Review by mslipper — APPROVED** [review]

**Comment by protolambda** [comment]
Rebased to resolve merge conflicts

**Comment by protolambda** [comment]
Deleted code cov comment since it wasn't updating anymore. Coverage is stable, no changes to actual code, and we merge all the coverage reports of different modules by pushing them all to codecov. You can find the coverage diff here: https://app.codecov.io/gh/ethereum-optimism/optimistic-specs/compare/main...go-work (some renames, basically 0 coverage difference)

**Review by trianglesphere — COMMENTED** [review]
I left some small comments around the makefile.
Also FYI - the integration test didn't pass locally, but I think that is probably the test being flaky b/c it passed in CI.

**Comment by protolambda** [comment]
I can help fix this merge conflict with a rebase in 4 hours from now. Basically need to rebase on `main`, drop all the old file moves + import rewrites, and redo those, then fix up some go imports, while preserving the docker and ci config fixes.

**Pull request closed without merging**